servers/physics_3d/godot_collision_solver_3d.cpp

```diff
@@ -587,3 +587,196 @@ bool GodotCollisionSolver3D::solve_distance(const GodotShape3D *p_shape_A, const
 		return gjk_epa_calculate_distance(p_shape_A, p_transform_A, p_shape_B, p_transform_B, r_point_A, r_point_B); //should pass sepaxis..
 	}
 }
+
+struct _ConcaveSweepInfo {
+	const Transform3D *transform_A = nullptr;
+	const GodotShape3D *shape_A = nullptr;
+	const Transform3D *transform_B = nullptr;
+	Vector3 original_motion_A;
+	Vector3 motion_A;
+	uint32_t tests = 0;
+	real_t min_advance = 0;
+	bool collided = false;
+	real_t error = 0;
+};
+
+bool GodotCollisionSolver3D::concave_sweep_callback(void *p_userdata, GodotShape3D *p_convex) {
+	_ConcaveSweepInfo &cinfo = *(static_cast<_ConcaveSweepInfo *>(p_userdata));
+	cinfo.tests++;
+
+	real_t advance = 0;
+
+	bool collided = sweep(cinfo.shape_A, *cinfo.transform_A, p_convex, *cinfo.transform_B, cinfo.motion_A, advance, AABB(), cinfo.error);
+
+	if (collided) {
+		if (!cinfo.collided || advance < cinfo.min_advance) {
+			cinfo.min_advance = advance;
+			cinfo.motion_A = cinfo.original_motion_A * advance; // Shorten distance for future collisions.
+		}
+
+		cinfo.collided = true;
+
+		if (advance == 0.0) {
+			return true; // No need to continue processing collisions, no further advance possible.
+		}
+	}
+
+	return false;
+}
+
+bool GodotCollisionSolver3D::sweep(const GodotShape3D *p_shape_A, const Transform3D &p_transform_A, const GodotShape3D *p_shape_B, const Transform3D &p_transform_B, const Vector3 &p_motion_A, real_t &r_distance, const AABB &p_concave_hint, real_t p_error) {
+	if (p_shape_A->is_concave()) {
```

Comment:
This test is bailing out if shape A is concave; however further down where the code handles B being concave, it also checks for A being concave again and bails out. I *think* this is a repeat of https://github.com/godotengine/godot/pull/70654. In that case we removed the initial concave bail-out because the subsequent code handled it correctly.